File: Atcoder_#355/javascript/b_piano2.js
function Main(input) {
  const lines = input.trim().split("\n");

  const [N, M] = lines[0].split(' ').map(Number);

  const A = lines[1].split(' ').map(Number);
  const B = lines[2].split(' ').map(Number);

  const C = A.concat(B).sort((a, b) => a - b);

  let consecutiveInA = false;
  for (let i = 0; i < C.length - 1; i++) {
    if (A.includes(C[i]) && A.includes(C[i + 1])) {
      consecutiveInA = true;
      break;
    }
  }

  if (consecutiveInA) {
    console.log("Yes");
  } else {
    console.log("No");
  }
}

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
Main(input);

File: Atcoder_#355/ruby/b_piano2.rb
n, m = gets.split.map(&:to_i)

a = gets.split.map(&:to_i)
b = gets.split.map(&:to_i)

c = (a + b).sort

consecutive_a = false
(n + m - 1).times do |i|
  if a.include?(c[i]) && a.include?(c[i + 1])
    consecutive_a = true
    break
  end
end

result = consecutive_a ? "Yes" : "No"

puts result
